olp-cpp-sdk-authentication/include/olp/authentication/AuthorizeRequest.h

```diff
@@ -0,0 +1,233 @@
+   * in the response is ALLOW.
+   * If the operator is 'and' or missing (defaults to 'and'),
+   * one of the following algorithms is applied:
+   * * If ANY actions have an individual policy decision of DENY,
+   *  the overall policy decision returned in the response is DENY.
+   * * If ALL actions have an individual policy decision of ALLOW,
+   * the overall policy decision returned in the response is ALLOW.
+   */
+  enum class DecisionApiOperatorType { kAnd, kOr };
+
+  /**
+   * @brief Gets the ID of the requested service.
+   *
+   * @return The service ID.
+   */
+  inline const std::string& GetServiceId() const { return service_id_; }
+
+  /**
+   * @brief Sets the service ID.
+   *
+   * @param service_id The service ID.
+   *
+   * @return A reference to the updated `DecisionRequest` instance.
+   */
+  inline AuthorizeRequest& WithServiceId(std::string service_id) {
+    service_id_ = std::move(service_id);
+    return *this;
+  }
+
+  /**
+   * @brief Gets the contract ID.
+   *
+   * @note: If the contract ID is not provided in the request, one of the
+   * following happens:
+   * * If you have permissions for a single contract ID associated with the
+   * requested service ID, the system automatically determines the contract ID.
+   * * If you have permissions for multiple contract IDs, the 'Contract
+   * Required' error is returned.
+   *
+   * @return The contract ID or an error.
+   */
+  inline const boost::optional<std::string>& GetContractId() const {
+    return contract_id_;
+  }
+
+  /**
+   * @brief Sets the contract ID.
+   *
+   * @see `GetContractId` for information on the contract ID.
+   *
+   * @param contract_id Your contract ID.
+   *
+   * @return A reference to the updated `DecisionRequest` instance.
+   */
+  inline AuthorizeRequest& WithContractId(
+      boost::optional<std::string> contract_id) {
+    contract_id_ = std::move(contract_id);
+    return *this;
+  }
+
+  /**
+   * @brief Sets the contract ID.
+   *
+   * @see `GetContractId` for information on the contract ID.
+   *
+   * @param contract_id Your contract ID.
+   *
+   * @return A reference to the updated `DecisionRequest` instance.
+   */
+  inline AuthorizeRequest& WithContractId(std::string contract_id) {
+    contract_id_ = std::move(contract_id);
+    return *this;
+  }
+
+  /**
+   * @brief Gets all actions.
+   *
+   * @return The vector of the actions (action-resorce pairs).
+   */
+  inline const Actions& GetActions() const { return actions_; }
+
+  /**
+   * @brief Adds the action-resource pair.
+   *
+   * @param[in] action The action that is used to determine an individual policy
+   * decision.
+   * @param[in] resource The resource that is used to request the decision for
+   * the action.
+   *
+   * @return A reference to the updated `DecisionRequest` instance.
+   */
+  inline AuthorizeRequest& WithAction(
+      std::string action, boost::optional<std::string> resource = boost::none) {
```

Comment:
Are you sure that the resource should be optional? Can it not be just a nomal string with users responsability to leave empty or fill with value.

Reply:
changed to empty string